torchgeo/datasets/geo.py

```diff
@@ -354,10 +301,10 @@ class RasterDataset(GeoDataset):
     date_format = '%Y%m%d'
 
     #: Minimum timestamp if not in filename
-    mint: float = 0
+    mint: datetime = pd.Timestamp.min
```

Comment:
Pandas defaults to `datetime64[ns]`, which has a much smaller range than `datetime`:
```pycon
>>> import pandas as pd
>>> pd.Timestamp.min
Timestamp('1677-09-21 00:12:43.145224193')
>>> pd.Timestamp.max
Timestamp('2262-04-11 23:47:16.854775807')
>>> datetime.min
datetime.datetime(1, 1, 1, 0, 0)
>>> datetime.max
datetime.datetime(9999, 12, 31, 23, 59, 59, 999999)
```
I can't imagine this being an issue, but something to be aware of.

Reply:
May actually convert the types directly to pd.Timestamp, datetime-compatibility is not great...